### PR TITLE
docker image was renamed to docker manifest

### LINKED ISSUE #3381: docker image renamed to docker manifest
hammer docker image subcommand was renamed to manifest.
(image was conflicting name within katello models...)

Let's adapt our automation to this change

### PR DESCRIPTION
**Pull request by lpramuk**
docker image was renamed to docker manifest

(Fixes #3381)

### CODE REVIEW

**Comment by lpramuk** [comment]
fixed CLI + API
for UI touched docstrings only

**Comment by elyezer** [comment]
ACK. Would be good to have some test results to make sure it is working fine.

**Comment by lpramuk** [comment]
I know, testing in progress

**Comment by lpramuk** [comment]
CLI Results
```
nosetests -v test_docker
...
Ran 36 tests in 1457.480s

FAILED (SKIP=2, errors=19)
```
Errors:
- Error: Unrecognised option '--lifecycle-environment-id' (10x)
- Unable to log in to this Docker Registry - getaddrinfo: Name or service not known (SocketError) (18x)
- Url URL must be valid and schema must be one of http and https (4x)

**Comment by lpramuk** [comment]
The last two errors are beyond scope of this PR (external registries testing needs refactoring)

I will address hammer option rename 
```--lifecycle-environment-id```  >> ```--to-lifecycle-environment-id```

**Comment by lpramuk** [comment]
@coveralls Can I get your ACK then? :-)

**Comment by elyezer** [comment]
Looking over the latest automation result for cli.test_docker it have run 25 tests, 14 fails and 11 passes.

This means that now we are running more tests right?

Also looking at you results analysis, there was 19 failures but summing all the numbers 10+18+4 gives more than what is shown there. Can you explain how you got that please?

File: robottelo/cli/docker.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def create(cls, options=None):
                                                       values.
             --registry-id REGISTRY_ID                 Registry this container
                                                       will have to use to get
-                                                      the image
+                                                      the manifest
```
Comment (oshtaier) [inline robottelo/cli/docker.py]: I am actually not see that information for latest SNAP, please update code correspondingly
```
>hammer docker container create --help
...
 --organizations ORGANIZATION_NAMES        Comma separated list of values.
 --registry REGISTRY_NAME                  Name to search by
 --registry-id REGISTRY_ID                  
 --repository-name REPOSITORY_NAME         Name of the repository to use to create the container. e.g. centos
 --tag TAG                                 Tag to use to create the container. e.g. latest
...
```

File: robottelo/cli/docker.py [hunk 2]
```diff
@@ -233,37 +233,37 @@ class DockerImage(Base):
...
 
     @classmethod
     def list(cls, options=None, per_page=True):
-        """List docker images
+        """List docker manifests
 
         Usage::
 
-            hammer docker image list [OPTIONS]
+            hammer docker manifest list [OPTIONS]
```
Comment (oshtaier) [inline robottelo/cli/docker.py]: I can see that `hammer docker manifest list` command has more option. Please, update options list if you like

**Comment by lpramuk** [comment]
Re: 10+18+4 > 19tests -  multiple errors per test

I plan to get results against master for better comparison

File: tests/foreman/cli/test_docker.py [hunk 3]
```diff
@@ -85,18 +85,18 @@ def test_positive_read_docker_tags(self):
...
             u'repository-id': repository['id'],
         })
-        # Some images do not have tags associated with it, ignore those because
-        # we want to check the tag information
-        images = [
-            image for image in images_list if int(image['tag-count']) > 0
+        # Some manifests do not have tags associated with it, ignore those
+        # because we want to check the tag information
+        manifests = [
+            m for m in manifests_list if int(m['tag-count']) > 0
```
Comment (oshtaier) [inline tests/foreman/cli/test_docker.py]: Please, avoid single character variable names

**Comment by elyezer** [comment]
I wondering if nose is reporting subTest results that could explain the numbers.

**Comment by oshtaier** [comment]
strong ACK from my side (pending small non-blocking comments)
that was so silly to have `lifecycle-environment-id` parameter for cv promote
Thanks @lpramuk for fixing that

File: robottelo/cli/docker.py (hunk 2 above)
Comment (lpramuk) [inline robottelo/cli/docker.py]: OK.

File: robottelo/cli/docker.py (hunk 1 above)
Comment (lpramuk) [inline robottelo/cli/docker.py]: OK.

**Comment by lpramuk** [comment]
6.1:
```# hammer --csv docker image list
ID,Image ID,Size,Tag count,Tags
```
vs.

6.2: 
```
# hammer --csv docker manifest list
ID,Name,Schema Version,Digest,Downloaded,Tag Name
```

m[tag-counts] wont definitely work. fixing....

**Comment by lpramuk** [comment]
I'm done. 
```
Ran 40 tests in 1544.649s

FAILED (SKIP=2, errors=16)
```

**Comment by lpramuk** [comment]
FYI: When updating AK I added lifecycle environment to avoid error:
```
Could not update the activation key:
  Error: found more than one lifecycle_environment
```
Code:
```
        ActivationKey.update({
            'id': activation_key['id'],
            'organization-id': self.org['id'],
            'content-view-id': another_cv['id'],
+           'lifecycle-environment-id': self.lce['id'],
        })
```

**Comment by lpramuk** [comment]
Yay, once enabled ```[docker]``` section it got yet better:
```
Ran 51 tests in 2281.221s

FAILED (SKIP=5, errors=14)
```

**Comment by elyezer** [comment]
ACK

**Comment by oshtaier** [comment]
ACK

**Pull request merged**